cmd/minishift/cmd/start.go

```diff
@@ -250,14 +250,25 @@ func runStart(cmd *cobra.Command, args []string) {
 		fmt.Println("-- OpenShift cluster will be configured with ...")
 		fmt.Println("   Version:", requestedOpenShiftVersion)
 
-		fmt.Printf("-- Copying oc binary from the OpenShift container image to VM ")
+		fmt.Printf("-- Pulling the Openshift Container Image ")
 		progressDots := progressdots.New()
 		progressDots.Start()
+		// We need to make sure if images are already exist from the cache then don't pull it again.
+		imageExist, err := dockerCommander.IsImageExist(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
+		if !imageExist {
+			_, err = dockerCommander.Pull(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
+			if err != nil {
+				atexit.ExitWithMessage(1, fmt.Sprintf("Error copying the oc binary to %s: %v", minishiftConstants.OcPathInsideVM, err))
+			}
+		}
+		progressDots.Stop()
+		fmt.Println(" OK")
+
+		fmt.Printf("-- Copying oc binary from the OpenShift container image to VM ...")
```

Comment:
The code around copying oc binary from the OpenShift container should be in a different function. I do not see any reason to keep the code in `runStart`.

Reply:
@LalatenduMohanty It is in a different function here we are using that function only in the runStart, what does mean it should be in a different function?

```golang
fmt.Printf("-- Copying oc binary from the OpenShift container image to VM ...")
err = clusterup.CopyOcBinaryFromImageToVM(dockerCommander, minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion), minishiftConstants.OcPathInsideVM)
 		if err != nil {
 			atexit.ExitWithMessage(1, fmt.Sprintf("Error copying the oc binary to %s: %v", minishiftConstants.OcPathInsideVM, err))
 		}

```

Reply:
> It is in a different function here we are using that function only in the runStart, what does mean it should be in a different function?

So I am talking these code .  We should put then in a function and call the function in runStart()

```code
fmt.Printf("-- Pulling the Openshift Container Image ")
		progressDots := progressdots.New()
		progressDots.Start()
		// We need to make sure if images are already exist from the cache then don't pull it again.
		imageExist, err := dockerCommander.IsImageExist(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
		if !imageExist {
			_, err = dockerCommander.Pull(minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion))
			if err != nil {
				atexit.ExitWithMessage(1, fmt.Sprintf("Error copying the oc binary to %s: %v", minishiftConstants.OcPathInsideVM, err))
			}
		}
		progressDots.Stop()
		fmt.Println(" OK")

		fmt.Printf("-- Copying oc binary from the OpenShift container image to VM ...")
		err = clusterup.CopyOcBinaryFromImageToVM(dockerCommander, minishiftConstants.GetOpenshiftImageName(requestedOpenShiftVersion), minishiftConstants.OcPathInsideVM)
		if err != nil {
			atexit.ExitWithMessage(1, fmt.Sprintf("Error copying the oc binary to %s: %v", minishiftConstants.OcPathInsideVM, err))
		}
		fmt.Println(" OK")
```

Reply:
@LalatenduMohanty Suggest which location you want to put and also the name, I will put it.